chia/full_node/full_block_utils.py

```diff
@@ -228,13 +248,14 @@ def generator_from_block(buf: memoryview) -> Optional[bytes]:
 @dataclass(frozen=True)
 class GeneratorBlockInfo:
     prev_header_hash: bytes32
+    height: uint32
     transactions_generator: Optional[SerializedProgram]
     transactions_generator_ref_list: list[uint32]
 
 
 def block_info_from_block(buf: memoryview) -> GeneratorBlockInfo:
     buf = skip_list(buf, skip_end_of_sub_slot_bundle)  # finished_sub_slots
-    buf = skip_reward_chain_block(buf)  # reward_chain_block
+    buf, height = height_from_reward_chain_block(buf)  # reward_chain_block
```

Comment:
We can avoid this added complexity by receiving `height` in `block_info_from_block` if we must (to keep returning this augmented `GeneratorBlockInfo`) instead of computing it. The caller has it already.